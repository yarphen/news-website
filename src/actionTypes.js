const keyMirror = obj => Object.keys(obj).reduce((prev, key) => ({ ...prev, [key]: key }), {});

export default keyMirror({
  PING_REQUEST_START: 0,
  PING_REQUEST_SUCCESS: 0,
  PING_REQUEST_FAIL: 0,
  PING_RESET: 0,
});
