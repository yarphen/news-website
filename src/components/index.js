export * from './ContentItem';
export * from './LoadMore';
export * from './Sidebar';
export * from './Header';
export * from './Footer';
export * from './ClearFilters';
export * from './Search';
