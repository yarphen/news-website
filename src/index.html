<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="manifest" href="/manifest.json" />
  <script async src="https://cdn.jsdelivr.net/npm/pwacompat@2.0.6/pwacompat.min.js"
    integrity="sha384-GOaSLecPIMCJksN83HLuYf9FToOiQ2Df0+0ntv7ey8zjUHESXhthwvq9hXAZTifA"
    crossorigin="anonymous"></script>
</head>

<body>
<div id="root">

</div>
</body>

</html>