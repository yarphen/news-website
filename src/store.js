import {
  applyMiddleware, compose, createStore, combineReducers,
} from 'redux';
import thunk from 'redux-thunk';
import { routerMiddleware, connectRouter } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { createLogger } from 'redux-logger';

import * as appReducers from './reducers';

const history = createBrowserHistory();

const middlewares = [thunk];

if (process.env.NODE_ENV === 'development') {
  middlewares.push(createLogger({ collapsed: true }));
}

const reducers = combineReducers({
  router: connectRouter(history), ...appReducers,
});

const enhancers = compose(applyMiddleware(
  routerMiddleware(history), ...middlewares,
));

const configureStore = (initialState = {}) => createStore(reducers, initialState, enhancers);

const store = configureStore();

export { store, history };
