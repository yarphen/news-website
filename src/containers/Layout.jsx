import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Sidebar, Header, Footer } from '../components';

class Layout extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { sidebarOpen: false };
    this.handleCloseSidebar = this.handleCloseSidebar.bind(this);
    this.handleOpenSidebar = this.handleOpenSidebar.bind(this);
  }

  handleCloseSidebar() {
    this.setState({ sidebarOpen: false });
  }

  handleOpenSidebar() {
    this.setState({ sidebarOpen: true });
  }

  render() {
    const { children, title } = this.props;
    const { sidebarOpen } = this.state;
    return (
      <div className="container">
        <Sidebar isOpen={sidebarOpen} onClose={this.handleCloseSidebar} />
        <Header onOpenSidebar={this.handleOpenSidebar} title={title} />
        {children}
        <Footer />
      </div>
    );
  }
}

Layout.propTypes = {
  children: PropTypes.any,
  title: PropTypes.string.isRequired,
};

Layout.defaultProps = {
  children: null,
};

export { Layout };
