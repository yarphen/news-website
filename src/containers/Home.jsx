import React, { PureComponent } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import PropTypes from 'prop-types';

import { pingActions } from '../actions';
import {
  selectorResult, selectorLoading, selectorInitial, selectorError,
} from '../selectors';
import { Layout } from './Layout';
import { ContentItem, LoadMore } from '../components';

class Home extends PureComponent {
  render() {
    // const { items } = this.props;
    const items = Array.from({ length: 6 }).map((item, index) => ({
      text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
        ullamco laboris nisi ut aliquip ex ea. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
        eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
        ullamco laboris nisi ut aliquip ex ea. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
        eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
        ullamco laboris nisi ut aliquip ex ea.`,
      title: 'Title',
      image: '',
      updated: new Date(),
      id: index,
    }));
    return (
      <Layout>
        {items.length && (
          <div className="content">
            {items.map(item => <ContentItem {...item} key={item.id} />)}
          </div>
        )}
        {items.length && <LoadMore />}
      </Layout>
    );
  }
}

Home.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object),
};

Home.defaultProps = {
  items: [],
};

const mapStateToProps = createSelector(
  [selectorResult, selectorLoading, selectorInitial, selectorError],
  (result, loading, initial, error) => ({
    result, loading, initial, error,
  }),
);

const mapDispatchToProps = dispatch => ({
  actions: {
    ping: bindActionCreators(pingActions, dispatch),
  },
});

const HomeWrapped = connect(mapStateToProps, mapDispatchToProps)(Home);

export { HomeWrapped as Home };
