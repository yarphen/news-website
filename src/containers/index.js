import { Home } from './Home';
import { withTag } from '../hoc';

export * from './Home';
export * from './NotFound';
export * from './Layout';

const News = withTag('news')(Home);
const Regions = withTag('regions')(Home);
const Video = withTag('video')(Home);
const TV = withTag('tv')(Home);

export {
  News,
  Regions,
  Video,
  TV,
};
