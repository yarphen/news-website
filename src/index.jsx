import React from 'react';
import ReactDOM from 'react-dom';
import { Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import { store, history } from './store';
import { MyContainer, NotFound } from './containers';

import './css/app.css';

ReactDOM.render(
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <Switch>
        <Route path="/" exact component={MyContainer} />
        <Route path="*" component={NotFound} />
      </Switch>
    </ConnectedRouter>
  </Provider>,
  document.getElementById('root'),
);
