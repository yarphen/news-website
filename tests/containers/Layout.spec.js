import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import { create } from 'react-test-renderer';
import { Layout } from '../../src/containers';
import { store } from '../../src/store';

describe('Layout container', () => {
  test('it matches the snapshot', () => {
    const container = create(
      <Provider store={store}><MemoryRouter><Layout /></MemoryRouter></Provider>,
    );
    expect(container.toJSON()).toMatchSnapshot();
  });
});
