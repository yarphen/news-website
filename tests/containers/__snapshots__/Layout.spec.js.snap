// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout container it matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="sidebar sidebar-hidden"
  >
    <div
      className="sidebar-inset"
    >
      <a
        className="sidebar-item"
        href="/news"
        onClick={[Function]}
      >
        News
      </a>
      <a
        className="sidebar-item"
        href="/regions"
        onClick={[Function]}
      >
        Regions
      </a>
      <a
        className="sidebar-item"
        href="/video"
        onClick={[Function]}
      >
        Video
      </a>
      <a
        className="sidebar-item"
        href="/tv"
        onClick={[Function]}
      >
        TV
      </a>
      <span
        className="sidebar-footer-item"
      >
        Copyright © yarphen
      </span>
    </div>
  </div>
  <div
    className="header-wrapper"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="big-header-logo fas fa-registered"
      />
    </a>
    <div
      className="mobile-header"
    >
      <div
        className="header-title"
      >
        News
      </div>
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="header-logo fas fa-registered"
        />
      </a>
      <div
        className="header-sandwich fas fa-bars"
        onClick={[Function]}
      />
    </div>
    <div
      className="nav-menu"
    >
      <a
        className="nav-menu-item"
        href="/news"
        onClick={[Function]}
      >
        News
      </a>
      <a
        className="nav-menu-item"
        href="/regions"
        onClick={[Function]}
      >
        Regions
      </a>
      <a
        className="nav-menu-item"
        href="/video"
        onClick={[Function]}
      >
        Video
      </a>
      <a
        className="nav-menu-item"
        href="/tv"
        onClick={[Function]}
      >
        TV
      </a>
      <div
        className="search"
      >
        <input
          className="search-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <div
          className="search-input-icon fas fa-search"
        />
      </div>
    </div>
  </div>
  <div
    className="footer-wrapper"
  >
    <div
      className="footer"
    >
      <div
        className="footer-left"
      >
        <div
          className="footer-left-item"
        >
          <div
            className="footer-left-item-wrapper"
          >
            <a
              href="/news"
              onClick={[Function]}
            >
              News
            </a>
          </div>
        </div>
        <div
          className="footer-left-item"
        >
          <div
            className="footer-left-item-wrapper"
          >
            <a
              href="/regions"
              onClick={[Function]}
            >
              Regions
            </a>
          </div>
        </div>
        <div
          className="footer-left-item"
        >
          <div
            className="footer-left-item-wrapper"
          >
            <a
              href="/video"
              onClick={[Function]}
            >
              Video
            </a>
          </div>
        </div>
        <div
          className="footer-left-item"
        >
          <div
            className="footer-left-item-wrapper"
          >
            <a
              href="/tv"
              onClick={[Function]}
            >
              TV
            </a>
          </div>
        </div>
      </div>
      <div
        className="footer-right"
      >
        Copyright © yarphen
      </div>
    </div>
  </div>
</div>
`;
